### PR TITLE
Add more behat scenarios for recurring payments

### PR DESCRIPTION
**Pull request by marekrzytki**

### CODE REVIEW

File: features/admin/creating_recurring_subscription_through_admin.feature [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  @ui @javascript
+  Scenario: Creating an order with Mollie Subscription payment for an existing customer
+    When I create a new order for "mail@example.com" and channel "United States"
+    And I add "Stark Coat" to this order
+    And I specify this order shipping address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
+    And I select "Mollie Subscription" payment method
+    And I place and confirm this order
+    Then I should be notified that order has been successfully created
+    And there should be one not paid nor shipped order with channel "United States" for "mail@example.com" in the registry
```
Comment (senghe) [inline features/admin/creating_recurring_subscription_through_admin.feature:24]: How about checking the payment type?

File: tests/Application/templates/bundles/SyliusAdminBundle/Order/Show/_payment.html.twig [hunk 2]
```diff
@@ -17,10 +17,28 @@
     {% if sm_can(payment, 'complete', 'sylius_payment') %}
         <div class="ui segment">
             <form action="{{ path('sylius_admin_order_payment_complete', {'orderId': order.id, 'id': payment.id}) }}" method="post" novalidate>
-                <input type="hidden" name="_csrf_token" value="{{ csrf_token(payment.id) }}" />
                 <input type="hidden" name="_method" value="PUT">
+                <input type="hidden" name="_csrf_token" value="{{ csrf_token(payment.id) }}" />
                 <button type="submit" class="ui icon labeled tiny blue fluid loadable button"><i class="check icon"></i> {{ 'sylius.ui.complete'|trans }}</button>
             </form>
         </div>
+        {% if payment.details['payment-link'] is defined %}
```
Comment (senghe) [inline tests/Application/templates/bundles/SyliusAdminBundle/Order/Show/_payment.html.twig:25]: Why this change? Shouldn't we add only tests?

File: tests/Behat/Context/Setup/SubscriptionContext.php [hunk 3]
```diff
@@ -45,8 +102,56 @@ public function thisOrderHasAnActiveMollieSubscription(OrderInterface $order): v
...
+        $this->entityManagerSubscription->persist($subscription);
         $this->entityManagerSubscription->flush();
     }
+
+    /**
+     * @When I run command :command
+     */
+    public function iRunCommand(string $command): void
+    {
+        $this->mollieApiClient->setApiEndpoint('http://localhost:8218');
```
Comment (senghe) [inline tests/Behat/Context/Setup/SubscriptionContext.php:144]: I think it's unnecessary as we have separate api client service for test environment

File: tests/Behat/Resources/services/contexts/setup.yml [hunk 4]
```diff
@@ -28,6 +28,9 @@ services:
         arguments:
```
Comment (senghe) [inline tests/Behat/Resources/services/contexts/setup.yml:28]: Why the `tests/Behat/Resources/services.xml` is an XML, but the current file is Yaml?

**Review by senghe — CHANGES_REQUESTED** [review]

File: tests/Application/templates/bundles/SyliusAdminBundle/Order/Show/_payment.html.twig (hunk 2 above)
Comment (marekrzytki) [reply tests/Application/templates/bundles/SyliusAdminBundle/Order/Show/_payment.html.twig:25]: It's part of installation AdminOrderCreation plugin, same as added Repositories

File: features/admin/create_scheduled_mollie_subscription.feature [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  Scenario: Recurring payment
+    When I go to "/"
+    When I browse orders
+    Then I should see a single order from customer "sylius@example.com"
+    When I run command "cd tests/Application && APP_ENV=test php bin/console mollie:subscription:begin-processing"
+    Then I should see "[OK] Successfully marked scheduled subscriptions" in the output
+    When I run command "cd tests/Application && APP_ENV=test php bin/console mollie:subscription:process"
+    Then I should see "[OK] Successfully marked scheduled subscriptions" in the output
+    When I browse orders
+    Then I should see 2 orders in the list
```
[the remark this answers is not in the document]
Comment (marekrzytki) [reply features/admin/create_scheduled_mollie_subscription.feature:32]: Done

File: features/admin/create_scheduled_mollie_subscription.feature [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    And all orders have same total set to "$1.00"
+
+  @ui
+  Scenario: Failed recurring payment
+    When this order is incomplete
+    And I go to "/"
+    When I run command "cd tests/Application && APP_ENV=test php bin/console mollie:subscription:begin-processing"
+    Then I should see "[OK] Successfully marked scheduled subscriptions" in the output
+    When I run command "cd tests/Application && APP_ENV=test php bin/console mollie:subscription:process"
+    Then I should see "[ERROR] An error has occurred during send payment link process." in the output
```
[the remark this answers is not in the document]
Comment (marekrzytki) [reply features/admin/create_scheduled_mollie_subscription.feature:68]: Done

File: features/admin/creating_recurring_subscription_through_admin.feature (hunk 1 above)
Comment (marekrzytki) [reply features/admin/creating_recurring_subscription_through_admin.feature:24]: Done

File: tests/Behat/Context/Setup/SubscriptionContext.php (hunk 3 above)
Comment (marekrzytki) [reply tests/Behat/Context/Setup/SubscriptionContext.php]: Sure, removed

File: tests/Behat/Resources/services/contexts/setup.yml (hunk 4 above)
Comment (marekrzytki) [reply tests/Behat/Resources/services/contexts/setup.yml:28]: It needs to stay that way as there is problem with paths in YAML, discussed with @senghe

**Review by senghe — APPROVED** [review]
